src/app/digital/models/ioobjects/other/ICData.ts

```diff
@@ -191,21 +191,26 @@ export class ICData {
         if (!this.IsValid(copies))
             return undefined;
 
-        // // Set designer of copies to null
-        // copies.toList().forEach((obj) => obj.setDesigner(undefined));
-
         // Move non-whitelisted inputs to regular components list
         //  So that the ports that come out of the IC are useful inputs and not
         //  things like ConstantHigh and ConstantLow which aren't interactive
         const INPUT_WHITELIST = [Switch, Button];
+        const OUTPUT_WHITELIST = [LED];
+
+        const inputs  = copies.getInputs().filter( i => INPUT_WHITELIST.some( (type) => i instanceof type));
+        const outputs = copies.getOutputs().filter(o => OUTPUT_WHITELIST.some((type) => o instanceof type));
+        const others  = copies.getComponents().filter(c => (!inputs.includes(c) && !outputs.includes(c)));
+
+        // Sort inputs/outputs by their position
+        const sortByPos = (a: Component, b: Component) => {
+            const p1 = a.getPos(), p2 = b.getPos();
+            if (Math.abs(p2.y - p1.y) <= 0.5*GRID_SIZE) // If same-ish-y, sort by x from LtR
+                return p2.x - p1.x;
+            return p2.y - p1.y; // Sort by y-pos from Top to Bottom
+        }
+        inputs.sort(sortByPos);
+        outputs.sort(sortByPos);
 
-        const inputs = copies.getInputs().filter((i) => INPUT_WHITELIST.some((type) => i instanceof type));
-        const others = copies.getOthers().concat(copies.getInputs())
-                .filter((c) => !INPUT_WHITELIST.some((type) => c instanceof type));
-
-        copies.setInputs(inputs);
-        copies.setOthers(others);
```

Comment:
These functions have a TODO commented above them marking them for deletion once no longer in use. This is the only place in the code where they are used, so the functions themselves could/should also be deleted.